services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js

```diff
@@ -261,6 +270,38 @@ export class NoticeOfWorkApplication extends Component {
     this.props.closeModal();
   };
 
+  handleUpdateLeadInspector = (finalThen) => {
+    this.props
+      .updateNoticeOfWorkApplication(
+        { lead_inspector_party_guid: this.state.associatedLeadInspectorPartyGuid },
+        this.props.noticeOfWork.now_application_guid,
+        `Successfully assigned ${
+          this.props.inspectorsHash[this.state.associatedLeadInspectorPartyGuid]
+        } as the Lead Inspector`
+      )
+      .then(() => {
+        this.props.fetchImportedNoticeOfWorkApplication(
+          this.props.noticeOfWork.now_application_guid
+        );
+      })
+      .then(() => finalThen());
```

Comment:
You can just call 'this.props.closeModal' here without passing it as a param

Reply:
Ah, I see what you did here. Disregard the above - can you update the name of this param - finalAction? I don't have a good recommendation here, `finalThen` is a little confusing.